packages/components/src/particles/Button/index.tsx

```diff
@@ -1,52 +1,36 @@
 import { clsx } from 'clsx';
+import { JSX, mergeProps } from 'solid-js';
 
-type ButtonVariant = 'primary' | 'secondary';
-type ButtonSize = 'xl' | 'lg' | 'md' | 'sm' | 'xs';
+export type ButtonVariant = 'primary' | 'secondary' | 'tertiary';
+export type ButtonSize = 'xl' | 'lg' | 'md' | 'sm' | 'xs';
 
-export interface ButtonProps {
-  onClick?: () => void;
-  disabled?: boolean;
+export interface ButtonProps extends JSX.ButtonHTMLAttributes<HTMLButtonElement> {
   variant?: ButtonVariant;
   size?: ButtonSize;
-  children: string;
 }
 
-export default function Button({
-  onClick,
-  disabled = false,
-  variant = 'primary',
-  size = 'md',
-  children
-}: ButtonProps) {
-  function handleClick() {
-    if (!disabled && onClick) {
-      onClick();
-    }
-  }
+export default function Button(props: ButtonProps) {
+  const merged = mergeProps({ variant: 'primary', size: 'lg' }, props);
 
-  const buttonClasses = clsx(
-    ['xl', 'lg', 'md'].includes(size) ? 'rounded-md' : 'rounded',
-    'font-sans box-border',
-    variant === 'primary' ? 'text-primaryText' : 'text-secondaryText',
-    variant === 'secondary' ? 'border-2 border-primary600' : 'border-2 border-transparent',
-    'shadow-sm',
-    'focus-visible:outline',
-    'focus-visible:outline-2',
-    'focus-visible:outline-offset-2',
-    variant === 'primary' ? 'bg-primary600' : 'bg-secondary50',
-    size === 'xl' && 'text-lg py-4 px-6',
-    size === 'lg' && 'text-base py-3 px-4',
-    size === 'md' && 'text-sm py-2 px-3',
-    size === 'sm' && 'text-xs py-1.5 px-2.5',
-    size === 'xs' && 'text-xs py-1 px-2',
-    disabled
-      ? 'opacity-50 cursor-not-allowed'
-      : 'hover:bg-primary500 focus-visible:outline-primary600'
-  );
+  const buttonClasses = clsx('font-semibold shadow-sm', {
+    'rounded-md': ['xl', 'lg', 'md'].includes(merged.size),
+    rounded: !['xl', 'lg', 'md'].includes(merged.size),
+    'text-white bg-indigo-600 hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 hover:bg-indigo-500':
+      merged.variant === 'primary',
+    'rounded bg-white text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 hover:bg-gray-50':
```

Comment:
How would we handle dynamic theming? (very future problem, but just thinking if it's worth preparing for, like centralizing colors or something)

Reply:
I set up the tailwind config, with a package[ tw-colors](https://www.npmjs.com/package/tw-colors). An earlier version shows how you can theme the buttons with this, and turns out to be pretty trivial change with find and replace text once we think through the variable setup. I paused on rolling this out until we get this first version not to slow things down and because there are a few tailwind theme libraries that are still low support, so I want to see how that changes over the next months before we go all in on one

https://github.com/Photon-Health/client/blob/438eec043904a7361ed136cc578a8c78f0612d38/packages/components/tailwind.config.cjs#L13